core/store/src/flat/store_helper.rs

```diff
@@ -57,11 +57,16 @@ pub fn remove_delta(store_update: &mut StoreUpdate, shard_uid: ShardUId, block_h
     store_update.delete(DBCol::FlatStateDeltaMetadata, &key);
 }
 
-pub fn remove_all_deltas(store_update: &mut StoreUpdate, shard_uid: ShardUId) {
+pub fn remove_range_by_shard_uid(
```

Comment:
this is potentially dangerous public API since it accepts any `DBCol` while not all of those are prefixed by `shard_uid`.
I suggest to keep that as a private helper function and expose 2 public functions:
```
pub(crate) fn remove_all_flat_state_values(store_update: &mut StoreUpdate, shard_uid: ShardUId) -> Result<..> {
    remove_range_by_shard_uid(store_update, shard_uid, &DBCol::FlatState)
}
pub(crate) fn remove_all_deltas(store_update: &mut StoreUpdate, shard_uid: ShardUId) ...
```

also I suggest changing parameter `cols: &[DBCol]` to `col: &DBCol` and just calling it twice in `remove_all_deltas` for changes and metadata.